src/main/java/Main.java

```diff
@@ -1,13 +1,71 @@
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Scanner;
 
 public class Main {
+    static List<Integer> numbers = new ArrayList<>();
```

Comment:
Problem polega na tym, że używasz statycznej listy (czyli należącej do klasy, a nie do obiektu). Testy tworzą za każdym razem nowy obiekt, ale ta lista pozostaje wspólna dla wszystkich, więc dane się nie usuwają. Zamień na niestatyczną i dodaj `private`

Reply:
Rozumiem. Nie ma pojęcia dlaczego oznaczyłem to pole jako statyczne.